src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -439,19 +440,208 @@ default boolean canSee(final Entity entity) {
     /**
      * {@link Keys#FIRE_DAMAGE_DELAY}
      *
-     * @return The amount of time to delay in ticks before an Entity will be burned by fire.
+     * @return The amount of time to delay in ticks before the entity will be burned by fire
      */
-    default Optional<Value.Mutable<Ticks>> fireImmuneTicks() {
-        return this.getValue(Keys.FIRE_DAMAGE_DELAY).map(Value::asMutable);
+    default Value.Mutable<Ticks> fireImmuneTicks() {
+        return this.requireValue(Keys.FIRE_DAMAGE_DELAY).asMutable();
     }
 
     /**
      * {@link Keys#TRANSIENT}
      *
      * @return The transient state
      */
-    default Optional<Value.Mutable<Boolean>> isTransient() {
-        return this.getValue(Keys.TRANSIENT).map(Value::asMutable);
+    default Value.Mutable<Boolean> isTransient() {
+        return this.requireValue(Keys.TRANSIENT).asMutable();
+    }
+
+    /**
+     * {@link Keys#AGE}
+     *
+     * @return The age of this entity
+     */
+    default Value.Mutable<Integer> age() {
+        return this.requireValue(Keys.AGE).asMutable();
+    }
+
+    /**
+     * {@link Keys#BASE_SIZE}
+     *
+     * @return The base size of the entity
+     */
+    default Value<Double> baseSize() {
+        return this.requireValue(Keys.BASE_SIZE);
+    }
+
+    /**
+     * {@link Keys#EYE_HEIGHT}
+     *
+     * @return The height of the eyes
+     */
+    default Value<Double> eyeHeight() {
```

Comment:
May be wrong but I don't think all entities have eye height and position.

Reply:
https://github.com/SpongePowered/Sponge/blob/1d33b3c3daf2fbcee44554d128deec16c11aa96e/src/main/java/org/spongepowered/common/data/provider/entity/EntityData.java#L78-L81